aws_cloudtrail_rules/aws_snapshot_made_public.py

```diff
@@ -1,20 +1,23 @@
+from panther_base_helpers import deep_get
+
+
 def rule(event):
     # EC2 Volume snapshot made public
-    if event['eventName'] == 'ModifySnapshotAttribute':
-        parameters = event.get('requestParameters', {})
-        if parameters.get('attributeType') != 'CREATE_VOLUME_PERMISSION':
+    if event.get("eventName") == "ModifySnapshotAttribute":
+        parameters = event.get("requestParameters", {})
+        if parameters.get("attributeType") != "CREATE_VOLUME_PERMISSION":
             return False
 
-        items = parameters.get('createVolumePermission',
-                               {}).get('add', {}).get('items', [])
+        items = deep_get(parameters, "createVolumePermission", "add", "items", default=[])
         for item in items:
-            if item.get('group') == 'all':
+            if not isinstance(item, dict):
```

Comment:
This is not correct - we don't return dicts cc @georgepsarakis . I think this should be `isinstance(item, Mapping)`

Reply:
Same goes for every use of `dict` in the rest of the rules.

Reply:
Good catch, thanks @kostaspap ! :)

Reply:
This is the only instance (besides another in the global helpers) where we do is isinstance check on `dict`

Debugging shows that the items are actually dicts, perhaps this is because items is set by:

`items = deep_get(parameters, "createVolumePermission", "add", "items", default=[])`

Maybe I'm on an older version of PAT? Unclear why this is different for me

@georgepsarakis maybe you could provide some insight here

In the mean time, I'll include the changes to check against `Mapping`

Reply:
PAT will work just fine, because it uses only Python builtins. The object passed to the rule function is a sub-class of `collections.abc.Mapping`, as well as its nested objects. `dict` is a sub-class of `Mapping` also, so everything should work with this check!

Reply:
> PAT will work just fine, because it uses only Python builtins. The object passed to the rule function is a sub-class of `collections.abc.Mapping`, as well as its nested objects. `dict` is a sub-class of `Mapping` also, so everything should work with this check!

Cool. I am a bit skeptical whenever I see instance checks with `dict`  since the latest issues :)

Reply:
True, it was a good catch.